### PR TITLE
Run babel twice if needed for ES modules

### PR DESCRIPTION
**Pull request by yardnsm**
In order for us to support tree-shaking, we need to run babel twice: for a CommonJS version, and for a ES modules version.

(This PR won't break existing projects)

When there's a `module` field in `package.json`, `yoshi-babel` will run babel twice:
- According to user's `.babelrc`. Output goes to `./dist/es`.
- According to user's `.babelrc` with [`transform-es2015-modules-commonjs`](https://babeljs.io/docs/plugins/transform-es2015-modules-commonjs/) plugin.

This requires users to add a `module` field to their `package.json` and modify their `.babelrc` to **not** transpile ES `import/export` to CommonJS `require`:

```diff
{
  "presets": [
-    "es2015",
+    ["es2015", { 
+      "modules": false
+    ],
    "react",
    "stage-2"
  ]
}
```

```diff
{
   "main": "./dist/src/index.js",
+   "module": "./dist/es/src/index.js",
}
```

The same goes to `babel-preset-env` and the `transform-es2015-modules-commonjs` plugin.

When in watch mode or when running tests, the `transform-es2015-modules-commonjs` plugin applies.

**EDIT:** since `yoshi-jest` does not use the require hook, when setting `modules: false` in `.babelrc`,  the tests are failing. In `wix-style-react`, the solution was as follows:

1. Install the `transform-es2015-modules-commonjs` plugin.
2. Install to latest version of `babel-jest`.
3. Add a `test` env to babel:
```json
"env": {
  "test": {
    "plugins": [
      "transform-es2015-modules-commonjs"
    ]
  }
}
```
```

### CODE REVIEW

**Comment by shahata** [comment]
didn't understand why we need `BABEL_ENV=test` if anyway `NODE_ENV` already equals `test`

**Comment by shahata** [comment]
also - merge it already :)

File: plugins/yoshi-babel/package.json [hunk 1]
```diff
@@ -28,7 +30,7 @@
     "mocha": "^3.2.0",
     "mocha-env-reporter": "^2.0.4",
     "strip-ansi": "~4.0.0",
-    "test-phases": "latest",
+    "test-phases": "^1.0.13",
```
Comment (tqmukas) [inline plugins/yoshi-babel/package.json]: I don't understand why you changed `test-phases` version.

File: plugins/yoshi-babel/package.json [hunk 2]
```diff
@@ -12,13 +12,15 @@
...
+    "babel-plugin-transform-es2015-modules-commonjs": "^6.24.1",
     "chalk": "^1.1.3",
     "gulp": "~3.9.1",
     "gulp-babel": "~6.1.2",
     "gulp-file-transform-cache": "~1.0.6",
     "gulp-plumber": "~1.1.0",
     "gulp-sourcemaps": "~2.6.0",
-    "mkdirp": "~0.5.1"
+    "mkdirp": "~0.5.1",
+    "shelljs": "^0.7.6"
```
Comment (tqmukas) [inline plugins/yoshi-babel/package.json]: Please do not use `shelljs` directly. Use `test-phases`.

File: plugins/yoshi-babel/test/helpers/hooks.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const sh = require('shelljs');
+
+module.exports = {
+  installDependencies: cwd => {
+    return exec('yarn install --no-lockfile', cwd);
+  }
+};
+
+function exec(cmd, cwd) {
```
Comment (tqmukas) [inline plugins/yoshi-babel/test/helpers/hooks.js]: Please use `test-phases`. Hooks serve for different purpose.

File: plugins/yoshi-babel/package.json (hunk 1 above)
Comment (yardnsm) [inline plugins/yoshi-babel/package.json]: I think it was the `start-runner sync`

**Pull request merged**